### PR TITLE
Edit Content: Explore the possibility to use web components for fields

### PR DESCRIPTION
**Pull request by rjvelazco**
### <samp>🤖 Generated by Copilot at 2c73542</samp>

This pull request adds two new apps to the `core-web` project: `contenttype-fields-builder` and `contenttype-fields-builder-e2e`. The former is an app that builds web components for content type fields using angular elements, and the latter is an app that tests the former using cypress. The pull request adds the configuration, source, and test files for both apps, as well as the eslint and typescript rules.

## Related Issue
Fixes #25434

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 2c73542</samp>

*  Add two new apps for end-to-end testing and web component building of content type fields ([link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-885b517bb9b69f05df7ea5bfe1fc79e91ffe4ebdc3484505457397d332796042R1-R17)-[link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-b10e916522c07345c45cdd172e92a745300fb8c83573fd474888774de1f54068R1-R22), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-a50c7f87ee9a9cab84261c1d83b53d6aa191c21eb084ce75ca8c23fbec795d12R1-R36)-[link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-e605b0dd02defe8dadb8cc0d657d12ea3b71feb18671b2bd06c9d86aab3a5b8aR1-R7))
  * Create configuration files for nx workspace, eslint, typescript, karma, and cypress for each app ([link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-885b517bb9b69f05df7ea5bfe1fc79e91ffe4ebdc3484505457397d332796042R1-R17), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-727a9154e6a346b52920ab63fb67b88962e53db421763f8d444855fe2fe12fd3R1-R12), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-1c620dac5f7e884a489414d8d816749e5b98cb8edab5ec251773771f0e8cf9aeR1-R31), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-b10e916522c07345c45cdd172e92a745300fb8c83573fd474888774de1f54068R1-R22), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-a50c7f87ee9a9cab84261c1d83b53d6aa191c21eb084ce75ca8c23fbec795d12R1-R36), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-be67612f92fc7ae5ba4e9b705fa8a8ff963765b707b2402d08c3f82b52ec584dR1-R16), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-8ef1779598dab26e24d0460334ae8b17856ddc5466ebb2cd125f59bb93137cffR1-R101), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-16577f630438af7ef513d00348e58f7812c8205c2a5bfac876e48d915fdd4d79R1-R10), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-e605b0dd02defe8dadb8cc0d657d12ea3b71feb18671b2bd06c9d86aab3a5b8aR1-R7))
  * Create entry point files for each app, such as `index.html`, `main.ts`, and `app.module.ts` ([link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769R1-R42), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-ef92bbc6d058cda69a2a5e04a42efbe3d058a822cefd45beb9af7a408fac003dR1-R11), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-72fea934369e1ad391b55bdb7b8860969dd296a5e33abb3735687c2ea3cefb69R1-R7))
  * Import the `ContenttypeFieldsModule` from the `@dotcms/contenttype-fields` library and define custom elements for web components in the `contenttype-fields-builder` app ([link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769R1-R42))
  * Create sample test and fixture files for the `contenttype-fields-builder-e2e` app, using cypress commands and helpers ([link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-f1f7b0aa07b7e3071c28afb887c7079fb560bc8505a5324b7ca49101ceb422d5R1-R4), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-f8f7315f39c206e6d9f8033d5b5e9866a142fdcdd5cabdfe4dbd10740ab7f4a7R1-R13))
  * Create helper and plugin files for the `contenttype-fields-builder-e2e` app, such as `app.po.ts`, `commands.ts`, and `index.js` ([link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-0cb75446e7e70bee53cdd9def788e6f3b5c3d4db82c6bd26c56bc6c6683c4e0aR1-R22), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-1181b65eb9cadf5d6d74e22c737c900e1fc969e98330b65f780cb90cf548b6ebR1), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-f5cb90fc8b87e50ff50d66944bf97228d9aef790fea114be5864487285099832R1-R33), [link](https://github.com/dotCMS/core/pull/25581/files?diff=unified&w=0#diff-91c01747fcf028cfae669988d14664890d7c67e5fe9adae762d198fd0efd9727R1-R17))

### CODE REVIEW

File: core-web/libs/contenttype-fields/karma.conf.js
Comment (zJaaal) [inline core-web/libs/contenttype-fields/karma.conf.js:1]: Shouldn't this have jest tests?

**Review by zJaaal — COMMENTED** [review]
See my comments

File: core-web/libs/contenttype-fields/karma.conf.js
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/karma.conf.js:1]: You are right! I'll change it

Comment (oidacra) [reply core-web/libs/contenttype-fields/karma.conf.js:1]: Yep.

**Pull request closed without merging**